src/main/kotlin/fi/metatavu/jsp/api/OrdersApiImpl.kt

```diff
@@ -63,6 +108,13 @@ class OrdersApiImpl: OrdersApi, AbstractApi() {
 
     override fun updateOrder(orderId: UUID, order: Order): Response {
         val existingOrder = ordersController.find(orderId) ?: return createNotFound("Order with id $orderId not found!")
+
```

Comment:
Function should update existing generic products, create added generic products and remove generic products that are no longer found in the updated order.